src/Common/src/TypeSystem/IL/Stubs/PInvokeMarshallingILEmitter.cs

```diff
@@ -100,6 +100,16 @@ private static bool IsBlittableStruct(TypeDesc type)
         {
             if (type.IsValueType)
             {
+                // Bool and Char are primitive types that are not simple types.
+                // Need to special case them because they have a field of the same
+                // type (primitive types are the only types allowed to do that).
+                // This would lead to recursion if not special cased.
+                TypeFlags category = type.Category;
+                if (category == TypeFlags.Boolean || category == TypeFlags.Char)
```

Comment:
Passing and returning blittable structs by value is always safe (unless there is marshalling directive to override the default).